File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        # length = rpointer - lpointer -1
        # breadth = min(height[lpointer], height[rpointer])
        # current_area = length * breadth
        # max_area = max(current_area, max_area)

        lpointer = 0
        rpointer = len(height) - 1
        max_area = 0

        while lpointer < rpointer:
            length = rpointer - lpointer
            breadth = min(height[lpointer], height[rpointer])

            current_area = length * breadth
            max_area = max(current_area, max_area)

            if height[lpointer] < height[rpointer]:
                lpointer += 1
            else:
                rpointer -=1
        
        return max_area

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int lPointer = 0;
        int rPointer = height.length - 1;
        int area = -1;
        int length;
        int breadth;
        int maxArea = 0;
        int lHeight;
        int rHeight;

        while(lPointer < rPointer) {
            length = rPointer - lPointer;
            lHeight = height[lPointer];
            rHeight = height[rPointer];

            // min height of wall
            if (lHeight < rHeight) {
                breadth = lHeight;
            } else {
                breadth = rHeight;
            }

            area = length * breadth;

            // setting the max Area
            if (area > maxArea) {
                maxArea = area;
            }

            // moving the bigger wall
            if (lHeight < rHeight){
                lPointer++;
            } else {
                rPointer--;
            }
        }
        
        return maxArea;
    }
}